src/index/visitor.rs

```diff
@@ -207,7 +207,6 @@ pub fn visit_pou(index: &mut Index, pou: &Pou) {
                 has_varargs,
                 pou.name_location.clone(),
             ));
-            index.register_init_function(&pou.name);
```

Comment:
Since we no longer track the init functions via map, I think we can combine `PouType::Function` and `PouType::Init | PouType::ProjectInit`  to a single match-variant